### PR TITLE
Cloud unit test

### PR DESCRIPTION
**Pull request by MorrisLaw**
**What this PR does / why we need it**:
This is an updated version of the original initial unit test PR (for cloud services and packages) created by Ahmad (rebased on main, go.mod updated, makefile updated). It was growing stale but I think it's still valuable and want to include as I work to update our tests. As stated in the original PR description this is the initial work for adding unit tests to cloud/services packages. The plan is to continue to add follow up PRs to include more unit tests. This PR gives us something to work with from now on.

**Special notes for your reviewer**:
Original PR from @prksu: https://github.com/kubernetes-sigs/cluster-api-provider-digitalocean/pull/235

I'll see if it passes the CI checks like before and I'll fix any failures as needed. It should pass the rebase one now. Once this is finished being reviewed and merged in I will close out the original PR.

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by cpanato** [comment]
cool!

there are some lint failures, can you take a look when you have some free cycle?
thanks

**Comment by MorrisLaw** [comment]
@cpanato It's ready for review again

File: cloud/services/computes/droplets_test.go [hunk 1]
```diff
@@ -0,0 +1,989 @@
+/*
+Copyright 2021 The Kubernetes Authors.
```
Comment (cpanato) [inline cloud/services/computes/droplets_test.go:2]: this is a new file the header should be 

```suggestion
Copyright 2023 The Kubernetes Authors.
```

File: cloud/services/computes/mock_computes/doc.go [hunk 2]
```diff
@@ -0,0 +1,25 @@
+/*
+Copyright 2021 The Kubernetes Authors.
```
Comment (cpanato) [inline cloud/services/computes/mock_computes/doc.go:2]: same here 
```suggestion
Copyright 2023 The Kubernetes Authors.
```

File: cloud/services/networking/loadbalancers_test.go [hunk 3]
```diff
@@ -0,0 +1,150 @@
+/*
+Copyright 2021 The Kubernetes Authors.
```
Comment (cpanato) [inline cloud/services/networking/loadbalancers_test.go:2]:
```suggestion
Copyright 2023 The Kubernetes Authors.
```

File: cloud/services/networking/mock_networking/doc.go [hunk 4]
```diff
@@ -0,0 +1,19 @@
+/*
+Copyright 2021 The Kubernetes Authors.
```
Comment (cpanato) [inline cloud/services/networking/mock_networking/doc.go:2]:
```suggestion
Copyright 2023 The Kubernetes Authors.
```

**Review by cpanato — CHANGES_REQUESTED** [review]
looks cool

a nit request change

File: cloud/services/networking/mock_networking/doc.go (hunk 4 above)
Comment (MorrisLaw) [reply cloud/services/networking/mock_networking/doc.go]: Good catch! Resolved those 4 instances of wrong year for newly added files @cpanato

**Review by cpanato — APPROVED** [review]
thank you!

**Pull request merged**